src/shader_recompiler/ir/operand_helper.cpp

```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "operand_helper.h"
+
+namespace Shader::IR {
+
+static IR::U32 GetBufferAddressComponent(const Inst* buffer_inst, u32 comp) {
```

Comment:
These functions are very small, it would be better to define them as inline in the header